common/rya.api/src/main/java/org/apache/rya/api/domain/serialization/kryo/RyaStatementSerializer.java

```diff
@@ -0,0 +1,156 @@
+package org.apache.rya.api.domain.serialization.kryo;
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import org.openrdf.model.impl.URIImpl;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.Serializer;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+
+import org.apache.rya.api.domain.RyaStatement;
+import org.apache.rya.api.domain.RyaType;
+import org.apache.rya.api.domain.RyaURI;
+
+public class RyaStatementSerializer extends Serializer<RyaStatement> {
+    
+    public static void writeToKryo(Kryo kryo, Output output, RyaStatement object) {
+        output.writeString(object.getSubject().getData());
+        output.writeString(object.getPredicate().getData());
+        output.writeString(object.getObject().getDataType().toString());
+        output.writeString(object.getObject().getData());
+        boolean hasContext = object.getContext() != null;
+        output.writeBoolean(hasContext);
+        if(hasContext){
+            output.writeString(object.getContext().getData());
+        }
+        boolean shouldWrite = object.getColumnVisibility() != null;
+        output.writeBoolean(shouldWrite);
+        if(shouldWrite){
+            output.writeInt(object.getColumnVisibility().length);
+            output.writeBytes(object.getColumnVisibility());
+        }
+        shouldWrite = object.getQualifer() != null;
+        output.writeBoolean(shouldWrite);
+        if(shouldWrite){
+            output.writeString(object.getQualifer());
+        }
+        shouldWrite = object.getTimestamp() != null;
+        output.writeBoolean(shouldWrite);
+        if(shouldWrite){
+            output.writeLong(object.getTimestamp());
+        }
+        shouldWrite = object.getValue() != null;
+        output.writeBoolean(shouldWrite);
+        if(shouldWrite){
+            output.writeBytes(object.getValue());
+        }
+    }   
+
+    @Override
+    public void write(Kryo kryo, Output output, RyaStatement object) {
+        writeToKryo(kryo, output, object);
+    }
+    
+    public static RyaStatement readFromKryo(Kryo kryo, Input input, Class<RyaStatement> type){
```

Comment:
This method an public static RyaStatement read(Input input) look identical. Why not just delegate to the other one since I think you're ignoring the kryo and type parameters.